src/common/kafka/serializer/Frame1DHistogramSerializer.h

```diff
@@ -0,0 +1,228 @@
+  static constexpr da00_dtype type = da00_dtype::uint8;
+};
+template <> struct data_type_trait<uint16_t> {
+  static constexpr da00_dtype type = da00_dtype::uint16;
+};
+template <> struct data_type_trait<uint32_t> {
+  static constexpr da00_dtype type = da00_dtype::uint32;
+};
+template <> struct data_type_trait<uint64_t> {
+  static constexpr da00_dtype type = da00_dtype::uint64;
+};
+template <> struct data_type_trait<float> {
+  static constexpr da00_dtype type = da00_dtype::float32;
+};
+template <> struct data_type_trait<double> {
+  static constexpr da00_dtype type = da00_dtype::float64;
+};
+
+/// @class Frame1DHistogramBuilder
+/// @brief A class that builds a 1D histogram frame for serialization.
+///
+/// \tparam T is the type of the data to be serialized
+/// \tparam R is the type of the data used for the axis. This can be time with
+/// double precession
+///
+/// The Frame1DHistogramBuilder class is responsible for building a 1D histogram
+/// frame for serialization. It provides methods to add data to the histogram,
+/// serialize the data, and initialize the X-axis values.
+template <class T, class R = T>
+class Frame1DHistogramSerializer : public AbstractSerializer {
+  using data_t = std::vector<std::vector<T>>;
+  using time_t = int32_t;
+
+  std::string _topic;
+  time_t _period;
+  time_t _binCount;
+  std::string _name;
+  std::string _unit;
+  std::string _timeUnit;
+  VectorAggregationFunc<T> aggregateFunction;
+
+protected:
+  data_t _data;
+  std::vector<R> _xAxis;
+
+public:
+  /// \brief Constructor for the Frame1DHistogramBuilder class.
+  ///
+  /// \param topic is the Kafka stream topic destination
+  /// \param period is the length of time of one frame in the specified units
+  /// \param binCount is the number of the bins used
+  /// \param name is the name of the binned data
+  /// \param unit is the unit of the binned data
+  /// \param timeUnit is the unit of time used for period, and the binned axis
+  /// \param callback is the producer callback function
+  /// \param aggFunc is the aggregation function used to aggregate the data
+  /// inide the bins
+  ///
+  Frame1DHistogramSerializer(
+      std::string topic, const time_t period, const time_t binCount,
+      std::string name, std::string unit,
+      std::string timeUnit =
+          "millisecond", //!< unit of time used for period, and the binned axis
+      const ProducerCallback &callback = {},
+      const VectorAggregationFunc<T> &aggFunc = SUM_AGG_FUNC<T>)
+      : AbstractSerializer(callback), _topic(std::move(topic)), _period(period),
+        _binCount(binCount), _name(std::move(name)), _unit(std::move(unit)),
+        _timeUnit(std::move(timeUnit)), aggregateFunction(aggFunc) {
+
+    initAxis();
+    _data = data_t(_xAxis.size());
+  }
+
+  inline void addData(const R rToA, const T data) {
+    static_assert(data_type_trait<R>::type != da00_dtype::none,
+                  "Data type R not supported for serialization!");
```

Comment:
Pease add here an config parameter which setup the binning strategy for data which out of the time scope, like add last bin or drop